### PR TITLE
feat: RAGEngine update and validation

### PR DESCRIPTION
**Pull request by bangqipropel**
**Reason for Change**:
Add Update part of RAGEngine controller and validation

**Requirements**

- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Notes for Reviewers**:

### CODE REVIEW

**Review by zhuangqh — APPROVED** [review]

File: pkg/ragengine/controllers/ragengine_controller.go [hunk 1]
```diff
@@ -152,8 +153,34 @@ func (c *RAGEngineReconciler) applyRAG(ctx context.Context, ragEngineObj *kaitov
 
 		if err = resources.GetResource(ctx, ragEngineObj.Name, ragEngineObj.Namespace, c.Client, deployment); err == nil {
 			klog.InfoS("An inference workload already exists for ragengine", "ragengine", klog.KObj(ragEngineObj))
-			return
+			if deployment.Annotations[kaitov1alpha1.RAGEngineRevisionAnnotation] != revisionStr {
+				var volumes []corev1.Volume
+				var volumeMounts []corev1.VolumeMount
+				shmVolume, shmVolumeMount := utils.ConfigSHMVolume(*ragEngineObj.Spec.Compute.Count)
```
Comment (Fei-Guo) [inline pkg/ragengine/controllers/ragengine_controller.go:159]: Do we need shmvolumes for rag engine?

File: pkg/ragengine/controllers/ragengine_controller.go [hunk 2]
```diff
@@ -152,8 +153,34 @@ func (c *RAGEngineReconciler) applyRAG(ctx context.Context, ragEngineObj *kaitov
 
 		if err = resources.GetResource(ctx, ragEngineObj.Name, ragEngineObj.Namespace, c.Client, deployment); err == nil {
 			klog.InfoS("An inference workload already exists for ragengine", "ragengine", klog.KObj(ragEngineObj))
-			return
+			if deployment.Annotations[kaitov1alpha1.RAGEngineRevisionAnnotation] != revisionStr {
```
Comment (Fei-Guo) [inline pkg/ragengine/controllers/ragengine_controller.go:156]: What are the fields that can be updated in ragengine cr?

File: pkg/ragengine/controllers/ragengine_controller.go (hunk 1 above)
Comment (ishaansehgal99) [reply pkg/ragengine/controllers/ragengine_controller.go:159]: I believe only if they use GPU node, and then run the embedding model locally on that

File: pkg/ragengine/controllers/ragengine_controller.go (hunk 2 above)
Comment (bangqipropel) [reply pkg/ragengine/controllers/ragengine_controller.go:156]: Until now, everything can be updated except the nodes

**Review by Fei-Guo — APPROVED** [review]

**Pull request merged**